exchange/bitswap/decision/engine.go

```diff
@@ -180,20 +180,19 @@ func (e *Engine) MessageReceived(p peer.ID, m bsmsg.BitSwapMessage) error {
 			log.Debug("wants", entry.Key, entry.Priority)
 			l.Wants(entry.Key, entry.Priority)
 			if exists, err := e.bs.Has(entry.Key); err == nil && exists {
-				newWorkExists = true
 				e.peerRequestQueue.Push(entry.Entry, p)
+				newWorkExists = true
```

Comment:
is there a strong reason to order these? if so maybe comment it? at first glance i wouldn't know and maybe screw up